lib/checkout.js

```diff
@@ -1,5 +1,10 @@
-const decodeAndParseCheckoutId = (encoded) => {
-  const decoded = Buffer.from(encoded, 'base64').toString('binary')
+const decodeAndParseCheckoutId = (input) => {
+  const isEncoded = input.slice(-1) === '='
```

Comment:
It's not the most sophisticated check 😅 🙈 

I tried several other solutions that the folks out there on the interwebz recommended, and none of them worked 😕 

It tried these:

```js
const isEncoded = Buffer.from(input, 'base64').toString('base64') === input
```

```js
const isEncoded = /^([0-9a-zA-Z+/]{4})*(([0-9a-zA-Z+/]{2}==)|([0-9a-zA-Z+/]{3}=))?$/.test(input)
```

```js
const isEncoded = input.length % 4 === 0 && /^[A-Za-z0-9+/]+[=]{0,3}$/.test(input)
```

```js
const isBase64 = (str) => {
    const notBase64 = /[^A-Z0-9+\/=]/i
    const len = str.length
    if (!len || len % 4 !== 0 || notBase64.test(str)) {
      return false
    }
    const firstPaddingChar = str.indexOf('=')
    return firstPaddingChar === -1 ||
      firstPaddingChar === len - 1 ||
      (firstPaddingChar === len - 2 && str[len - 1] === '=')
  }
```

Reply:
let's try it, we can iterate on it later

One way could be to try to match it to shopify's checkout id format

Reply:
This seems to be an easy and good solution

https://gist.github.com/RyanNutt/fe1836843a4257a4a362b2f551213835

Reply:
@istvankovacs yup, I saw that, that's basically the same as my first attempt:

```js
const isEncoded = Buffer.from(input, 'base64').toString('base64') === input
```

It's not reliable, as you can see [in the comments](https://gist.github.com/RyanNutt/fe1836843a4257a4a362b2f551213835#gistcomment-3495733), and it doesn't work for our actual text fixtures either